### DISCUSSION TITLE
Regex - escape characters.

### DISCUSSION THREAD

**Discussion by MariasStory** [sepinf-inc/IPED#2260] [General]
I am trying to use regex in search field. The problem is that I cannot get regexes working. One main issue that I see now is that the backslash is not working. Normally in a regex \. means a plane point ".". But in IPED it does not work.
**Please explain how to work with the regex in a search field.**
On another hand, simply modifying the RegexConfig.txt file did not work.
**Can you please explain how to get my own regexes working?**

PS: Thanks again for the cool solution.

**Comment by lfcnassif** [comment]
Regexes in the **index** search field only work if your regex contains only indexed chars, which by default are letters and numbers. You can add more chars to be indexed and allowed in the index search regexes into conf/IndexTaskConfig.txt->extraCharsToIndex. It must be configured before processing the case. Another limitation of those regexes is that they work intra terms, they do not cross word boundaries, eg, non indexed chars.

In the other hand, RegexConfig.txt allows for very flexible regexes, with arbitrary chars, modifying it before processing the case should work. But it's not a perl regex, it has a few limitations paid for very high speed. Its main limitation is that it doesn't support back references and named groups AFAIK. Please take a look at the dk brics automaton library documentation for its syntax: https://www.brics.dk/automaton/doc/dk/brics/automaton/RegExp.html
